File: app/src/main/java/ng/assist/UIs/Utils/LoadingDialogUtils.java
package ng.assist.UIs.Utils;

import android.app.Dialog;
import android.content.Context;
import android.widget.TextView;


import ng.assist.R;

public class LoadingDialogUtils {

    private Dialog loadingDialog;
    private TextView loadingText;
    private Context mContext;

    public LoadingDialogUtils(Context context){
        this.mContext = context;
        loadingDialog = new Dialog(mContext);
        loadingDialog.setContentView(R.layout.loading_dialog);
        loadingDialog.setCancelable(false);
        loadingText = loadingDialog.findViewById(R.id.loading_dialog_text);
    }

    public void showLoadingDialog(String text){
        loadingText.setText(text);
        loadingDialog.show();
    }
    public void cancelLoadingDialog(){
        loadingDialog.dismiss();
    }

}

File: app/src/main/java/ng/assist/UIs/Utils/LoadingDialogUtils.kt
package ng.assist.UIs.Utils

import android.app.Dialog
import android.content.Context
import android.widget.TextView


import ng.assist.R

class LoadingDialogUtils(private val mContext: Context) {

    private val loadingDialog: Dialog
    private val loadingText: TextView

    init {
        loadingDialog = Dialog(mContext)
        loadingDialog.setContentView(R.layout.loading_dialog)
        loadingDialog.setCancelable(false)
        loadingText = loadingDialog.findViewById(R.id.loading_dialog_text)
    }

    fun showLoadingDialog(text: String) {
        loadingText.text = text
        loadingDialog.show()
    }

    fun cancelLoadingDialog() {
        loadingDialog.dismiss()
    }

}
